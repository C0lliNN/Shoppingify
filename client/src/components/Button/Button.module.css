.Button {
  background-color: #303030;
  outline: none;
  border: none;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  color: white;
  cursor: pointer;
}

.Button:hover {
  background-color: #2b2b2b;
}

.Primary {
  background-color: rgb(0, 119, 255);
}

.Primary:hover {
  background-color: rgb(4, 108, 226);
}

.Success {
  background-color: rgb(14, 156, 90);
}
.Success:hover {
  background-color: rgb(14, 138, 80);
}

.Danger {
  background-color: rgb(255, 0, 64);
}

.Danger:hover {
  background-color: rgb(231, 0, 58);
}
