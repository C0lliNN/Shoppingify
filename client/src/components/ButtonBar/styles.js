import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

export const ButtonBarWrapper = styled.div`
  padding: 20px 0px;
  display: block;
  background: transparent;
  text-align: center;

  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    position: fixed;
    bottom: 0;
    right: 0;
    background: #fefefe;
    width: ${variables.INFO_BAR_MD_SIZE}px;
  }
  @media (min-width: ${variables.LG_BREAK_POINT}px) {
    width: ${variables.INFO_BAR_LG_SIZE}px;
  }
`;