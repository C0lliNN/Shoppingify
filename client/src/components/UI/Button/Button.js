import styled from 'styled-components';
import {
  COLORS,
  BORDER_RADIUS_1,
  BORDER_RADIUS_2,
  FONT_FAMILY,
  FONT_SIZE_2,
  BUTTON_TYPES,
} from '../../../helpers/style-constants';
import PropTypes from 'prop-types';

const Button = styled.button`
  ${(props) => {
    switch (props.btnType) {
      case 'raised': {
        return `
          background-color: ${COLORS[props.variant]}; 
          border-color: ${COLORS[props.variant]};
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.white}; 
          padding: 20px;
        `;
      }
      case 'outlined': {
        return `
          background-color: transparent;
          border-radius: ${BORDER_RADIUS_2}; 
          border-color: ${COLORS[props.variant]};
          color: ${COLORS[props.variant]};
          padding: 6px 15px;
        `;
      }
      case 'flat': {
        return `
          background-color: transparent;
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.black_2};
          border-color: transparent;
          padding: 20px;
        `;
      }
      default:
        return `
          background-color: ${COLORS[props.variant]}; 
          border-color: ${COLORS[props.variant]};
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.white}; 
          padding: 20px;
        `;
    }
  }}
  border-width: 2px;
  border-style: solid;
  font-size: ${FONT_SIZE_2};
  font-family: ${FONT_FAMILY};
  font-weight: bold;
  outline: none;
  cursor: pointer;
  &:disabled {
    ${(props) => {
      switch (props.btnType) {
        case 'raised': {
          return `
            background-color: ${COLORS.gray_2};
          `;
        }
        case 'outlined': {
          return `
            color: ${COLORS.gray_2}; 
          `;
        }
        case 'flat': {
          return `
            color: ${COLORS.gray_2}
          `;
        }
        default:
          return `
        background-color: ${COLORS.gray_2};
        `;
      }
    }}
  
    border-color: ${COLORS.gray_2};  
    cursor: default;
  }
`;

Button.propTypes = {
  btnType: PropTypes.oneOf(BUTTON_TYPES).isRequired,
  variant: PropTypes.oneOf(Object.keys(COLORS)),
};

export default Button;
