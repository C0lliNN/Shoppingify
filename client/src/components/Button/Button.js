import PropTypes from 'prop-types';
import React from 'react';
import styles from './Button.module.css';

export default function Button(props) {
  const classes = [styles.Button];

  switch (props.type) {
    case 'primary':
      classes.push(styles.Primary);
      break;
    case 'success':
      classes.push(styles.Success);
      break;
    case 'danger':
      classes.push(styles.Danger);
      break;
    default:
      classes.push(styles.Primary);
  }

  return (
    <button onClick={props.onClick} className={classes.join(' ')}>
      {props.children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['primary', 'success', 'danger']),
};
