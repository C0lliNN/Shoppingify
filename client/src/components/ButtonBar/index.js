import PropTypes from 'prop-types';
import React from 'react';
import { ButtonBarWrapper } from './styles';

export default function ButtonBar(props) {
  return <ButtonBarWrapper {...props}>{props.children}</ButtonBarWrapper>;
}

ButtonBar.propTypes = {
  children: PropTypes.any,
};
