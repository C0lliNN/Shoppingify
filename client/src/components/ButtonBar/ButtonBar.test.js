import React from 'react';
import { render } from '@testing-library/react';
import ButtonBar from './ButtonBar';
import Button from '../UI/Button/Button';
import 'jest-styled-components';

describe('<ButtonBar/>', () => {
  it('should render correctly', () => {
    const { container, getByText } = render(
      <ButtonBar>
        <Button btnType="raised" variant="primary">
          OK
        </Button>
      </ButtonBar>
    );

    expect(getByText('OK')).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
