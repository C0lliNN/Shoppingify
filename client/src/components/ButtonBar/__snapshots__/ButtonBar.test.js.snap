// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBar/> should render correctly 1`] = `
<div>
  <div
    class="sc-AxjAm jqnyKw"
  >
    <button
      class="sc-AxirZ cMsEQX"
    >
      OK
    </button>
  </div>
</div>
`;
